docs/aws/audit/iamcompliance/rules/root_account_mfa_remediation.mdx

```diff
@@ -1,6 +1,245 @@
+
+   ```sh
+   aws iam list-mfa-devices --user-name root
+   ```
+
+4. **Enforce MFA Usage:**
+   Optionally, you can create an IAM policy that enforces the use of MFA for sensitive operations. Attach this policy to the root account or other IAM users as needed.
+
+   ```sh
+   aws iam create-policy --policy-name EnforceMFA --policy-document '{
+       "Version": "2012-10-17",
+       "Statement": [
+           {
+               "Effect": "Deny",
+               "Action": "*",
+               "Resource": "*",
+               "Condition": {
+                   "BoolIfExists": {
+                       "aws:MultiFactorAuthPresent": "false"
+                   }
+               }
+           }
+       ]
+   }'
+   ```
+
+By following these steps, you can ensure that the root account in AWS IAM has MFA enabled, thereby enhancing the security of your AWS environment.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of not having Multi-Factor Authentication (MFA) enabled for the root account in AWS IAM using Python scripts, you can follow these steps:
+
+1. **Install AWS SDK for Python (Boto3):**
+   Ensure you have Boto3 installed in your Python environment. You can install it using pip if you haven't already.
+
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Python Script to Check MFA Status:**
+   Write a Python script that uses Boto3 to check if MFA is enabled for the root account. This script will help you identify if the root account does not have MFA enabled.
+
+   ```python
+   import boto3
+
+   def check_root_mfa():
+       client = boto3.client('iam')
+       response = client.get_account_summary()
+       mfa_devices = client.list_mfa_devices(UserName='root')
+       
+       if response['SummaryMap']['AccountMFAEnabled'] == 1 and len(mfa_devices['MFADevices']) > 0:
+           print("MFA is enabled for the root account.")
+       else:
+           print("MFA is NOT enabled for the root account. Please enable it.")
+
+   if __name__ == "__main__":
+       check_root_mfa()
+   ```
+
+3. **Automate the Script Execution:**
+   Schedule the script to run at regular intervals using a task scheduler like cron (Linux) or Task Scheduler (Windows) to ensure continuous monitoring.
+
+   For example, to run the script every day at midnight using cron, you can add the following line to your crontab:
+
+   ```bash
+   0 0 * * * /usr/bin/python3 /path/to/your_script.py
+   ```
+
+4. **Notify Administrators:**
+   Enhance the script to send notifications (e.g., via email or Slack) if MFA is not enabled. This ensures that administrators are alerted immediately and can take action.
+
+   ```python
+   import boto3
+   import smtplib
+   from email.mime.text import MIMEText
+
+   def send_notification(message):
+       # Set up the server and login details
+       smtp_server = 'smtp.example.com'
+       smtp_port = 587
+       smtp_user = 'your_email@example.com'
+       smtp_password = 'your_password'
```

Comment:
<h2>Secret Detected: Secret Keyword</h2> <a href="https://console.cloudanix.com/code?t=secrets"> Click here for more details</a>